views/archive.twig

```diff
@@ -1,10 +1,14 @@
 {% extends "base.twig" %}
 
 {% block content %}
-  <!-- Provided by views/archive.twig -->
-	<section>
-		<h1 class="page-title">{{__('Archives: ', 'gesso') }} {{ wp_title }}</h1>
-		{% include 'templates/loop.twig' %}
-		{% include 'templates/pager.twig' %}
-	</section>
+    <!-- Provided by views/archive.twig -->
+    <section class="section section--archive">
+        <div class="section__title">
```

Comment:
It's not clear what the purpose/difference is between the section, article and post classes in all of these files.  For example, in 404.twig the page title is article__title, but here the same thing is section__title.  I think all of these files need refactoring, but before diving into that you'll need to clearly plan out what defines the classes used (e.g., article, section, post) with an eye towards simplicity, semantics, and avoiding the need for code duplication.